content/src/service/pointers/PointerManager.ts

```diff
@@ -12,72 +9,38 @@ export class PointerManager {
    * Commit a new entity, and return the 'before & after' generated by the deployment
    */
   async referenceEntityFromPointers(
-    lastDeployedPointersRepo: LastDeployedPointersRepository,
+    deploymentsRepo: DeploymentsRepository,
     deploymentId: DeploymentId,
-    entity: Entity
+    entity: Entity,
+    overwrote: Set<number>,
+    overwritten: boolean
   ): Promise<DeploymentResult> {
-    // Fetch active last deployments on pointers
-    const lastDeployments = await lastDeployedPointersRepo.getLastActiveDeploymentsOnPointers(
-      entity.type,
-      entity.pointers
-    )
-
-    // Add a made up deployments for the pointers where there was no deployment yet
-    const pointersWithDeployments = lastDeployments
-      .map((deployment) => deployment.pointers)
-      .reduce((accum, curr) => accum.concat(curr), [])
-    const pointersWithoutDeployments = diff(entity.pointers, pointersWithDeployments)
-    if (pointersWithoutDeployments.size > 0) {
-      lastDeployments.push({
-        entityId: 'NOT_GONNA_BE_USED',
-        deployment: 0,
-        timestamp: -1,
-        pointers: Array.from(pointersWithoutDeployments.values()),
-        deleted: true
-      })
-    }
-
-    // Determine if the entity being deployed will become active
-    const willDeploymentBecomeActive = lastDeployments.every((lastDeployment) => happenedBefore(lastDeployment, entity))
-
-    // Prepare variables
     const result: DeploymentResult = new Map()
-    const overwrite: Set<string> = new Set()
 
-    lastDeployments.forEach((lastDeployment) => {
-      // Calculate the intersection of pointers between the last deployment and the new one
-      const intersection: Set<string> = intersect(lastDeployment.pointers, entity.pointers)
+    if (!overwritten) {
+      // Fetch overwritten deployments from the DB
+      const overwrittenDeployments = await deploymentsRepo.getDeployments(overwrote)
```

Comment:
Is it correct to name this `overwritten` if we are querying by `overwrote`?

Reply:
Renamed and documented